src/Node/Expression/CallExpression.php

```diff
@@ -52,7 +52,7 @@ protected function compileCallable(Compiler $compiler)
                     // Compile a non-optimized call to trigger a \Twig\Error\RuntimeError, which cannot be a compile-time error
                     $compiler->raw(\sprintf('$this->env->getExtension(\'%s\')', $class));
                 } else {
-                    $compiler->raw(\sprintf('$this->extensions[\'%s\']', ltrim($class, '\\')));
+                    $compiler->raw(\sprintf('$this->getExtension(\'%s\')', ltrim($class, '\\')));
```

Comment:
I wonder, why aren't we always using `$this->env->getExtension`? That one already has proper typing.

Reply:
maybe this was a performance optimization to avoid extra method calls (based on the comment a file lines above).

This could actually be supported for the existing code if https://github.com/phpstan/phpstan/issues/9521 was implemented in phpstan, as the type of `Template::$extensions` is `class-string-map<T of ExtensionInterface, T>`

Reply:
Yes, unfortunately we don't have it yet.